core/src/main/java/io/substrait/relation/Expand.java

```diff
@@ -17,8 +17,11 @@ public abstract class Expand extends SingleInputRel {
   @Override
   public Type.Struct deriveRecordType() {
     Type.Struct initial = getInput().getRecordType();
-    return TypeCreator.of(initial.nullable())
-        .struct(Stream.concat(initial.fields().stream(), Stream.of(TypeCreator.REQUIRED.I64)));
+    var fields =
+        getFields().isEmpty()
+            ? initial.fields().stream()
+            : Stream.concat(initial.fields().stream(), getFields().get(0).getTypes());
```

Comment:
this also doesn't seem right (but possible I'm not reading this right!) - [the spec](https://github.com/substrait-io/substrait/blob/2e13d0b7b4a2b5ac8221e7f02cc8f04456c8f11c/proto/substrait/algebra.proto#L475) says 
> In addition to a field being emitted per input field an extra int64 field is emitted which
> contains a zero-indexed ordinal corresponding to the duplicate definition.

so I'd expect the record type to be something like `initial.fields().map([for each field handle nullability based on duplicates]) + Type.Int64`